packages/core/src/services/essentials/events.ts

```diff
@@ -0,0 +1,30 @@
+// Internal Imports
+import { Service } from "../service";
+
+export class Events extends Service {
+	// DOM Event target
+	documentFragment: DocumentFragment;
+
+	init() {
+		// Setup the event fragment on the DOM
+		this.documentFragment = document.createDocumentFragment();
+	}
+
+	getDocumentFragment() {
```

Comment:
Instead of having each component using this service get the doc fragment directly, would it make sense to have a `addEventListener(event, handler)` method that handles adding event listeners? As well as a `removeEventListener` method ... that way we can change the underlying event bus (if needed) without disrupting all the components and consumers

Reply:
@cal-smith the `EventTarget` constructor seems to not be fully cross-browser compatible
![image](https://user-images.githubusercontent.com/14989804/67119704-b0da9380-f1b5-11e9-98e0-7ad413de6b75.png)

Reply:
That's entirely orthogonal to my suggestion 👍 You can still implement the same methods without extending `EventTarget`